packages/patternfly-4/content/design-guidelines/usage-and-behavior/modal/design.md

```diff
@@ -21,24 +21,80 @@ The PatternFly components listed in the following sections can be used in a numb
+
+**Note:** Since confirmation dialogs disrupt user task flow, do not provide confirmation dialogs for actions that are easily reversed or insignificant.
+
+##### Confirm a non-destructive action
+![Modal](img/confirm-non-destructive-action.png)
+Provide buttons on confirmation dialogs that make it simple for users to understand and confirm their choices. When choosing button labels, use descriptive verbs that clearly indicate the consequence of a selection, like Delete, Save, or Create. Avoid vague labels like OK or Done.
+
+Ambiguous title and button labels make it more difficult for the user to understand their options and make a decision. Descriptive title and button labels make it easy for the user to scan the content and make a decision quickly.
+
+##### Confirm a destructive action
+![Modal](img/confirm-destructive-action.png)
+If an action is difficult or impossible to undo a warning icon can be incorporated into the modal to add emphasis. If an action will seriously impact the system and cannot be undone, use a destructive style button. Otherwise, use a primary button.
+
+Sometimes it is necessary to require an additional confirmation with a destructive action. An example using a form field can be seen below. In order for the destructive action to become actionable, the user needs to type in a word or phrase to make the button active.
+![Modal](img/2-step-confirm-destructive-action.png)
+
+#### Report Errors
+![Modal](img/error-dialog.png)
+Use error dialogs to inform users of problems interrupting normal or expected behavior.
+An error message should be actionable, human readable, and answer the following questions:
+* What is the problem?
+* Why did it happen?
+* How can I fix it?
+
+Never blame the user. Use passive voice where appropriate to avoid assigning blame.
+* ⛔ You put in the wrong password.
+* ✅ That password wasn’t recognized OR
+* ✅ We didn’t recognize that password.
+
+#### Provide the result of a task or state of an application
+![Modal](img/passive-dialog.png)
+Use passive dialogs to communicate critical information like the status of an application or result of an action. The content of a passive dialog should be important, immediately relevant, and either lead to or require action on the user’s part.
+
+Some usage examples include:
+* Notifying a user that an operation may take a long time to complete
+* Notifying a user that a critical process is complete
+
+**Note:** If information is not critical, don’t use a passive dialog. Instead, use inline notifications or toast notifications.
+
+#### Get required user input
+Sometimes users may need to provide additional input in order to complete an action. In this situation a form or other type of element may be incorporated into the modal. An example of this use case is the [Modal Wizard](/documentation/core/demos/wizard).
```

Comment:
Not sure if I think this reference is needed here without more examples. @stacymcauliffe what do you think. I don't feel strongly about it, but seems that this specific document is now focused on modal messages and alerts.